diff --git a/system/core/adb/adb.c b/system/core/adb/adb.c
index 10a1e0da26..e7a887b4a3 100644
--- a/system/core/adb/adb.c
+++ b/system/core/adb/adb.c
@@ -1272,7 +1272,7 @@ static int should_drop_privileges() {
     */
     property_get("ro.kernel.qemu", value, "");
     if (strcmp(value, "1") != 0) {
-        property_get("ro.secure", value, "1");
+        property_get("ro.secure", value, "0");
         if (strcmp(value, "1") == 0) {
             // don't run as root if ro.secure is set...
             secure = 1;
