diff --git a/handlers/Config.in b/handlers/Config.in
index 1137acd..f6db62d 100644
--- a/handlers/Config.in
+++ b/handlers/Config.in
@@ -5,6 +5,15 @@
 
 menu "Image Handlers"
 
+config DISKPART
+	bool "disk partition"
+	default n
+	help
+	  Supporit GPT/DOS partition for mmc/sd devices.
+	  this will require 'parted' utils.
+	  Say Y if you have mmc/sd or you use disk on
+	  your system.
+
 config UBIVOL
 	bool "ubivol"
 	default n
@@ -198,4 +207,11 @@ config UCFWHANDLER
 comment "Microcontroller handler depends on libgpiod"
 	depends on !HAVE_LIBGPIOD
 
+config ANDROID_SPARSE_IMAGE
+	bool "Android sparse image"
+	default n
+	help
+	  This is a android sparse image handler that android sparse image copy
+	  into the destination.
+
 endmenu
diff --git a/handlers/Makefile b/handlers/Makefile
index 8db9e41..934cb57 100644
--- a/handlers/Makefile
+++ b/handlers/Makefile
@@ -19,3 +19,5 @@ obj-$(CONFIG_SHELLSCRIPTHANDLER) += shell_scripthandler.o
 obj-$(CONFIG_SWUFORWARDER_HANDLER) += swuforward_handler.o
 obj-$(CONFIG_UBIVOL)	+= ubivol_handler.o
 obj-$(CONFIG_UCFWHANDLER)	+= ucfw_handler.o
+obj-$(CONFIG_ANDROID_SPARSE_IMAGE) += libsparse/ android_sparse_image_handler.o
+obj-$(CONFIG_DISKPART)	+= diskpart_handler.o
