diff --git a/handlers/Config.in b/handlers/Config.in
index 41eac1c..56aed03 100644
--- a/handlers/Config.in
+++ b/handlers/Config.in
@@ -5,6 +5,15 @@
 
 menu "Image Handlers"
 
+config DISKPART
+	bool "disk partition"
+	default n
+	help
+	  Supporit GPT/DOS partition for mmc/sd devices.
+	  this will require 'parted' utils.
+	  Say Y if you have mmc/sd or you use disk on
+	  your system.
+
 config UBIVOL
 	bool "ubivol"
 	default n
@@ -246,4 +255,11 @@ config UCFW_OLD_LIBGPIOD
 	  Rather there is no way to get this changes from the library
 	  at build time.
 
+config ANDROID_SPARSE_IMAGE
+	bool "Android sparse image"
+	default n
+	help
+	  This is a android sparse image handler that android sparse image copy
+	  into the destination.
+
 endmenu
diff --git a/handlers/Makefile b/handlers/Makefile
index 61e4f76..b446520 100644
--- a/handlers/Makefile
+++ b/handlers/Makefile
@@ -21,3 +21,5 @@ obj-$(CONFIG_SSBLSWITCH) += ssbl_handler.o
 obj-$(CONFIG_SWUFORWARDER_HANDLER) += swuforward_handler.o swuforward-ws.o
 obj-$(CONFIG_UBIVOL)	+= ubivol_handler.o
 obj-$(CONFIG_UCFWHANDLER)	+= ucfw_handler.o
+obj-$(CONFIG_ANDROID_SPARSE_IMAGE) += libsparse/ android_sparse_image_handler.o
+obj-$(CONFIG_DISKPART)	+= diskpart_handler.o
