diff --git a/handlers/Makefile b/handlers/Makefile
index 61e4f76..350580b 100644
--- a/handlers/Makefile
+++ b/handlers/Makefile
@@ -10,7 +10,7 @@
 obj-y	+= dummy_handler.o
 obj-$(CONFIG_ARCHIVE) += archive_handler.o
 obj-$(CONFIG_BOOTLOADERHANDLER) += boot_handler.o
-obj-$(CONFIG_CFI)	+= flash_handler.o
+obj-$(CONFIG_CFI)	+= flash_handler.o mtd_handler.o
 obj-$(CONFIG_CFIHAMMING1)+= flash_hamming1_handler.o
 obj-$(CONFIG_LUASCRIPTHANDLER) += lua_scripthandler.o
 obj-$(CONFIG_RAW)	+= raw_handler.o
