From b9febd9ee7cd5df1302fe4a42a1aa31422bfbd8e Mon Sep 17 00:00:00 2001
From: suker <suker@nexell.co.kr>
Date: Wed, 12 Sep 2018 11:20:08 +0900
Subject: [PATCH] bl2 link error - __aeabi_idiv

---
 src/debug.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/debug.c b/src/debug.c
index be002bc..298af10 100644
--- a/src/debug.c
+++ b/src/debug.c
@@ -111,12 +111,17 @@ cbool DebugInit(u32 ch)
 	pReg_Uart->LCR_H = 0x0070;	// 8 bit, none parity, stop 1, normal mode
 	pReg_Uart->CR = 0x0300;		// rx, tx enable
 
+#if 0
 	pReg_Uart->IBRD =
 		(u16)((SOURCE_CLOCK / SOURCE_DIVID) /
 				((BAUD_RATE / 1) * 16)); // IBRD = 8, 115200bps
 	pReg_Uart->FBRD = (u16)((((((SOURCE_CLOCK / SOURCE_DIVID) %
 					((BAUD_RATE / 1) * 16)) + 32) * 64) /
 					((BAUD_RATE / 1) * 16))); // FBRD = 0,
+#else
+	pReg_Uart->IBRD = 8;
+	pReg_Uart->FBRD = 0;
+#endif
 	pReg_UartClkGen->clkenb =
 		(1 << 3) | (1 << 2);	// PCLKMODE : always, Clock Gen Enable
 	pReg_Uart->CR = 0x0301;		// rx, tx, uart enable
-- 
2.17.1

