From 3895fb9d5866bbd0acefb93ce6c796b9e45c9ff7 Mon Sep 17 00:00:00 2001
From: choonghyun-Jeon <suker@nexell.co.kr>
Date: Thu, 21 Jul 2016 14:25:23 +0900
Subject: [PATCH] optee os compile error patch

---
 mk/gcc.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mk/gcc.mk b/mk/gcc.mk
index fc38c4d..bf6323e 100644
--- a/mk/gcc.mk
+++ b/mk/gcc.mk
@@ -12,7 +12,7 @@ nostdinc$(sm)	:= -nostdinc -isystem $(shell $(CC$(sm)) \
 			-print-file-name=include 2> /dev/null)
 
 # Get location of libgcc from gcc
-libgcc$(sm)  	:= $(shell $(CC$(sm)) $(comp-cflags$(sm)) \
+#libgcc$(sm)  	:= $(shell $(CC$(sm)) $(comp-cflags$(sm)) \
 			-print-libgcc-file-name 2> /dev/null)
 
 # Define these to something to discover accidental use
-- 
2.7.4

