From 90899f37e076db0d9dbaae944c5e84022d2eca7b Mon Sep 17 00:00:00 2001
From: choonghyun-Jeon <suker@nexell.co.kr>
Date: Mon, 18 Jul 2016 13:18:32 +0900
Subject: [PATCH] NAVI NSP4330 Bl1 Board SETTING

---
 config.mak | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/config.mak b/config.mak
index 60df4ef..e87e77a 100644
--- a/config.mak
+++ b/config.mak
@@ -21,8 +21,8 @@ VERINFO				= V062
 ###########################################################################
 # Build Environment
 ###########################################################################
-#CHIPNAME			= NXP4330
-CHIPNAME			= S5P4418
+CHIPNAME			= NXP4330
+#CHIPNAME			= S5P4418
 
 DEBUG				= n
 
@@ -45,15 +45,15 @@ SUPPORT_NAND_BOOT		= n
 SUPPORT_UART_BOOT		= n
 
 ifeq ($(CHIPNAME), NXP4330)
-BOARD				= LEPUS
-#BOARD				= NAVI
+#BOARD				= LEPUS
+BOARD				= NAVI
 else
 #BOARD				= SVT
 #BOARD				= ASB
 #BOARD				= DRONE
 #BOARD				= AVN
 #BOARD				= LAVENDA
-BOARD				?= RAPTOR
+#BOARD				?= RAPTOR
 endif
 
 # cross-tool pre-header
-- 
2.7.4

